src/account/SemiModularAccount.sol

```diff
@@ -0,0 +1,180 @@
+    function setFallbackSignerDisabled(bool isDisabled) external wrapNativeFunction {
+        SemiModularAccountStorage storage _storage = _getSemiModularAccountStorage();
+        emit FallbackSignerDisabledSet(_storage.fallbackSignerDisabled, isDisabled);
+
+        _storage.fallbackSignerDisabled = isDisabled;
+    }
+
+    /// @notice Returns whether the fallback signer validation is disabled.
+    /// @return True if the fallback signer validation is disabled, false if it is enabled.
+    function isFallbackSignerDisabled() external view returns (bool) {
+        return _getSemiModularAccountStorage().fallbackSignerDisabled;
+    }
+
+    /// @notice Returns the fallback signer associated with this account, regardless if the fallback signer
+    /// validation is enabled or not.
+    /// @return The fallback signer address, either overriden in storage, or read from bytecode.
+    function getFallbackSigner() external view returns (address) {
+        return _getFallbackSigner();
+    }
+
+    function _execUserOpValidation(
+        ModuleEntity userOpValidationFunction,
+        PackedUserOperation memory userOp,
+        bytes32 userOpHash
+    ) internal override returns (uint256) {
+        if (userOpValidationFunction.eq(_FALLBACK_VALIDATION)) {
+            address fallbackSigner = _getFallbackSigner();
+
+            if (
+                SignatureChecker.isValidSignatureNow(
+                    fallbackSigner, userOpHash.toEthSignedMessageHash(), userOp.signature
+                )
+            ) {
+                return _SIG_VALIDATION_PASSED;
+            }
+            return _SIG_VALIDATION_FAILED;
+        }
+
+        return super._execUserOpValidation(userOpValidationFunction, userOp, userOpHash);
+    }
+
+    function _execRuntimeValidation(
+        ModuleEntity runtimeValidationFunction,
+        bytes calldata callData,
+        bytes calldata authorization
+    ) internal override {
+        if (runtimeValidationFunction.eq(_FALLBACK_VALIDATION)) {
+            address fallbackSigner = _getFallbackSigner();
+
+            if (msg.sender != fallbackSigner) {
+                revert FallbackSignerMismatch();
+            }
+            return;
+        }
+        super._execRuntimeValidation(runtimeValidationFunction, callData, authorization);
+    }
+
+    function _exec1271Validation(ModuleEntity sigValidation, bytes32 hash, bytes calldata signature)
+        internal
+        view
+        override
+        returns (bytes4)
+    {
+        if (sigValidation.eq(_FALLBACK_VALIDATION)) {
+            address fallbackSigner = _getFallbackSigner();
+
+            if (SignatureChecker.isValidSignatureNow(fallbackSigner, hash, signature)) {
+                return _1271_MAGIC_VALUE;
+            }
+            return _1271_INVALID;
+        }
+        return super._exec1271Validation(sigValidation, hash, signature);
+    }
+
+    function _globalValidationAllowed(bytes4 selector) internal view override returns (bool) {
```

Comment:
Do we also want to include `setFallbackSignerDisabled` here? If so, we should test these ability to call these functions.

Reply:
Do you mean testing it? Absolutely-- I should have included that here, but I'm starting work on a whole testing PR addressing some gaps here and there too.

Or are you referring to a specific scenario, like bricking the account?

Reply:
Oh by testing I just meant including some run where it uses `setFallbackSignerDisabled` to true in a test, so we can detect regressions if anything breaks in the future. And sounds good!

Reply:
So you think it's worth having a test scenario where an SMA is deployed, a new validation is installed and the fallback signer is disabled?

Reply:
Yes a test like that would help. The thing I wanted to call out was that it looks like that function is missing from line 141 - it should be included there, right?

Reply:
Ah sorry just understood, yeah good catch!

Reply:
Done in [dbbca2a](https://github.com/erc6900/reference-implementation/pull/123/commits/dbbca2a5ab59d52da93c29f912c01389d38d2b3e)